Objects/typevarobject.c

```diff
@@ -443,45 +443,42 @@ static PyMethodDef typevar_methods[] = {
 PyDoc_STRVAR(typevar_doc,
 "Type variable.\n\
 \n\
-Usage::\n\
-\n\
-  T = TypeVar('T')  # Can be anything\n\
-  A = TypeVar('A', str, bytes)  # Must be str or bytes\n\
+The preferred way to construct a type variable is via the dedicated syntax\n\
+for :ref:`generic functions <generic-functions>`,\n\
+:ref:`generic classes <generic-classes>`, and\n\
+:ref:`generic type aliases <generic-type-aliases>`::\n\
 \n\
-Type variables exist primarily for the benefit of static type\n\
-checkers.  They serve as the parameters for generic types as well\n\
-as for generic function definitions.  See class Generic for more\n\
-information on generic types.  Generic functions work as follows:\n\
+   class Sequence[T]:  # T is a TypeVar\n\
+         ...\n\
 \n\
-  def repeat(x: T, n: int) -> List[T]:\n\
-      '''Return a list containing n references to x.'''\n\
-      return [x]*n\n\
+This syntax can also be used to create bound and constrained type\n\
+variables::\n\
 \n\
-  def longest(x: A, y: A) -> A:\n\
-      '''Return the longest of two strings.'''\n\
-      return x if len(x) >= len(y) else y\n\
+   class StrSequence[S: str]:  # S is a TypeVar bound to str\n\
+         ...\n\
 \n\
-The latter example's signature is essentially the overloading\n\
-of (str, str) -> str and (bytes, bytes) -> bytes.  Also note\n\
-that if the arguments are instances of some subclass of str,\n\
-the return type is still plain str.\n\
 \n\
-At runtime, isinstance(x, T) and issubclass(C, T) will raise TypeError.\n\
+   class StrOrBytesSequence[A: (str, bytes)]:  # A is a TypeVar constrained to str or bytes\n\
+         ...\n\
 \n\
-Type variables defined with covariant=True or contravariant=True\n\
-can be used to declare covariant or contravariant generic types.\n\
-See PEP 484 for more details. By default generic types are invariant\n\
-in all type variables.\n\
+However, if desired, reusable type variables can also be constructed\n\
+manually, like so::\n\
 \n\
-Type variables can be introspected. e.g.:\n\
+   T = TypeVar('T')  # Can be anything\n\
+   S = TypeVar('S', bound=str)  # Can be any subtype of str\n\
+   A = TypeVar('A', str, bytes)  # Must be exactly str or bytes\n\
 \n\
-  T.__name__ == 'T'\n\
-  T.__constraints__ == ()\n\
-  T.__covariant__ == False\n\
-  T.__contravariant__ = False\n\
-  A.__constraints__ == (str, bytes)\n\
+Type variables exist primarily for the benefit of static type\n\
+checkers.  They serve as the parameters for generic types as well\n\
+as for generic function and type alias definitions.\n\
 \n\
-Note that only type variables defined in global scope can be pickled.\n\
+The variance of type variables is inferred by type checkers when they are created\n\
+through the type parameter syntax and when\n\
+``infer_variance=True`` is passed.\n\
```

Comment:
Looks weird in the REPL to have such short lines here

```suggestion
through the type parameter syntax and when ``infer_variance=True`` is passed.\n\
```